### ISSUE TITLE
`generator-superset` plugin uses deprecated and out of sync packages with the rest of Superset

### ISSUE THREAD

**Issue by wsoeltz** [apache/superset#27496]
### Bug description

The `generator-superset` plugin for creating custom data visualization plugins utilizes the `superset-ui` version of `chart-controls` and `core`, but these are now 3 years out of date with main. Are there current versions of these packages that can be swapped out in place of them? Otherwise we would need some to be made available in order to be able to continue making custom visualizations that don't fall out of sync with the rest of Superset.

https://github.com/apache/superset/blob/master/superset-frontend/packages/generator-superset/generators/plugin-chart/templates/package.erb#L32-L33
```erb
  "peerDependencies": {
    "@airbnb/config-babel": "^2.0.1",
    "@superset-ui/chart-controls": "*",
    "@superset-ui/core": "*",
    "react": "^16.13.1"
  },
```

### How to reproduce the bug

1. Run the `generator-superset` plugin
2. View the versions of @superset-ui/chart-controls and @superset-ui/core that are installed
3. Both are derived from the archived [superset-ui repo](https://github.com/apache-superset/superset-ui)

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Checklist

- [ ] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [ ] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [ ] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by musfir-tariq** [comment]
Hi @wsoeltz. Were you able to find a fix for this issue?

**Comment by wsoeltz** [comment]
> Hi @wsoeltz. Were you able to find a fix for this issue?

No, we ended up having to fork Superset and build our chart package inside the monorepo

**Comment by rusackas** [comment]
I'll leave this open for now, as it's likely true. The generator is gathering dust, and we intend to replace it with newer build tooling as part of the extensions initiative. That said, this current workflow will likely be officially deprecated soon. If anyone wants to update the generator package, that'd be more than welcomed as a contribution.

Also, if anyone has cool charts they want to share, we're always looking to learn about new ones!

**Comment by rusackas** [comment]
Closing this as not-planned, but we're always open to PRs bumping dependencies to fix this kind of thing, while we design and build the next generation of tooling for this.
